joblib/hashing.py

```diff
@@ -194,7 +194,10 @@ def save(self, obj):
                 # https://github.com/numpy/numpy/issues/4983.  The
                 # workaround is to view the array as bytes before
                 # taking the memoryview
-                obj_bytes_view = obj.view(self.np.uint8)
+                if obj.flags.f_contiguous:
+                    obj_bytes_view = obj.T.view(self.np.uint8).T
+                else:
+                    obj_bytes_view = obj.view(self.np.uint8)
```

Comment:
@lesteve, in fact there was a genuine issue in my code as the `self._hash.update()̀` function was expected a c_contiguous array (thanks for helping).
I pushed an updated version of the code which should be better now.